### PR TITLE
vstart: Support deployment of ganesha daemon by cephadm with NFS option

### PR DESCRIPTION
**Pull request by varshar16**
NFS Ganesha cluster can be created in two ways with vstart
cluster:

1) With test_orchestrator
NFS=1 ../src/vstart.sh
This type of deployment can have multiple ganesha daemons on random port.

2) With Cephadm
NFS=1 ../src/vstart.sh --cephadm
It can deploy only single ganesha daemon with vstart on default ganesha port.

Both can create multiple cephfs exports.

Fixes: https://tracker.ceph.com/issues/45830
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
ack.

**Comment by alfonsomthd** [comment]
jenkins test dashboard backend

**Comment by jmolmo** [comment]
Maybe i miss something ... but i do not understand well this change.
the thing to solve is:
"vstart: Support deployment of ganesha daemon by cephadm with NFS option"

and in the code modifications i only can see a Ceph nfs setup with the "test_orchestrator" enabled.
Cephadm orchestrator is not used here in any place, and for sure that a nfs daemon is not deployed with cephadm... (which is the thing that it seems the bug title suggest to do)

**Comment by jmolmo** [comment]
> Maybe i miss something ... but i do not understand well this change.
> the thing to solve is:
> "vstart: Support deployment of ganesha daemon by cephadm with NFS option"
> 
> and in the code modifications i only can see a Ceph nfs setup with the "test_orchestrator" enabled.
> Cephadm orchestrator is not used here in any place, and for sure that a nfs daemon is not deployed with cephadm... (which is the thing that it seems the bug title suggest to do)

ok. I think i know what is the motivation... just trying to simulate the cephadm behaviour.

File: src/pybind/mgr/test_orchestrator/dummy_data.json [hunk 1]
```diff
@@ -369,6 +396,20 @@
       "status": 1,
       "status_desc": "running",
       "version": "16.0.0-827-g61ad12e"
+    },
+    {
+      "container_id": "aeba86ca1655",
+      "container_image_id": "36114e38494190b0c9d4b088c12e6e4086e8017b96b4d5fc14eb5406bd51b55b",
+      "container_image_name": "quay.io/ceph-ci/ceph:master",
+      "created": "2020-04-16T05:44:41.551646",
+      "daemon_id": "nfs.ganesha-vstart.osd0",
```
[the remark this answers is not in the document]
Comment (varshar16) [reply src/pybind/mgr/test_orchestrator/dummy_data.json]: Done

File: src/pybind/mgr/test_orchestrator/dummy_data.json [hunk 2]
```diff
@@ -369,6 +396,20 @@
...
+      "daemon_type": "nfs",
+      "hostname": "osd0",
+      "last_refresh": "2020-04-16T06:51:43.182937",
+      "started": "2020-04-16T05:44:41.606514",
+      "status": 1,
+      "status_desc": "running",
+      "version": "16.0.0-827-g61ad12e"
```
[the remark this answers is not in the document]
Comment (varshar16) [reply src/pybind/mgr/test_orchestrator/dummy_data.json]: Fixed.

File: src/pybind/mgr/test_orchestrator/dummy_data.json [hunk 3]
```diff
@@ -311,6 +311,33 @@
...
+          {
+            "hostname": "osd0",
+            "name": "",
+            "network": ""
+          }
+        ]
+      },
+      "service_id": "ganesha-vstart",
+      "service_name": "nfs.ganesha-vstart",
+      "service_type": "nfs",
```
[the remark this answers is not in the document]
Comment (varshar16) [reply src/pybind/mgr/test_orchestrator/dummy_data.json:332]: Done

**Comment by varshar16** [comment]
> > Maybe i miss something ... but i do not understand well this change.
> > the thing to solve is:
> > "vstart: Support deployment of ganesha daemon by cephadm with NFS option"
> > and in the code modifications i only can see a Ceph nfs setup with the "test_orchestrator" enabled.
> > Cephadm orchestrator is not used here in any place, and for sure that a nfs daemon is not deployed with cephadm... (which is the thing that it seems the bug title suggest to do)
> 
> ok. I think i know what is the motivation... just trying to simulate the cephadm behaviour.

You are partially correct.

If `cephadm` is enabled then volume's nfs interface calls orchestrator to deploy ganesha daemons.
Otherwise as you said `test_orchestrator` simulates cephadm behaviour.

**Comment by ajarr** [comment]
@batrick this PR is good to merge. I don't think this PR can be tested in teuthology. It works for me locally with vstart.

**Comment by batrick** [comment]
jenkins test dashboard

**Pull request merged**